dspace-server-webapp/src/main/java/org/dspace/app/rest/link/HalLinkFactory.java

```diff
@@ -43,11 +43,8 @@ public List<Link> getLinksFor(HALResource halResource, Pageable pageable) throws
         return list;
     }
 
-
     protected <T> Link buildLink(String rel, T data) {
-        UriComponentsBuilder uriComponentsBuilder = uriBuilder(data);
-
-        return buildLink(rel, uriComponentsBuilder.build().toUriString());
+        return linkTo(data).withRel(rel);
```

Comment:
I'm not entirely sure, but this small change _might_ be the cause to the templated links issue I described in my feedback.